src/Prototype/Exe/Form/Signup.hs

```diff
@@ -0,0 +1,160 @@
+{-# LANGUAGE DeriveAnyClass #-}
+module Prototype.Exe.Form.Signup
+  ( Input
+  , Page(..)
+  , ResultPage(..)
+  ) where
+
+import           Smart.Html.Dsl                 ( HtmlCanvas )
+import qualified Smart.Html.Dsl                as Dsl
+import qualified Smart.Html.Render             as Render
+import           Smart.Html.Shared.Html.Icons
+import           Text.Blaze                     ( customAttribute )
+import qualified Text.Blaze.Html5              as H
+import           Text.Blaze.Html5               ( (!) )
+import qualified Text.Blaze.Html5.Attributes   as A
+import           Web.FormUrlEncoded             ( FromForm(..)
+                                                , parseUnique
+                                                )
+
+
+--------------------------------------------------------------------------------
+data Input = Input
+  { username     :: Text
+  , emailAddress :: Text
+  , password     :: Text
+  , iUnderstand  :: Text -- TODO Bool
+  }
+  deriving (Eq, Show)
+
+instance FromForm Input where
+  fromForm f =
+    Input
+      <$> parseUnique "username"     f
+      <*> parseUnique "email"        f
+      <*> parseUnique "password"     f
+      <*> parseUnique "i-understand" f
```

Comment:
If we're ok using the camel case field `iUnderstand` (in HTML) for  the consent checkbox; we can remove this boilerplate with Generic derivations, as above. 

I'm also strongly in favour of using the same field names as in Haskell in HTML, since it keeps the field mappings obvious.